### PR TITLE
Integrate Slack App with Pinecone

### PR DESCRIPTION
**Pull request by 20001LastOrder**
The `bolt_app` uses the official slack_bolt library supported to build Slack app with Python. The tool integrates the slack app with the Pinecone database by default, but it can be switched to a local vector database as well by switching the retrieval of the app from `vectorstores.py`

**Features**:
1. Respond to mentions and reply in a thread
2. The thread history is considered when conversation continues in the same thread
3. The Pinecone index is preloaded with documentation for different LLM-related libraries (see `scripts/Pinecone.ipynb`)
4. Each response is correspond with a reference on which library documentation was refered.

### CODE REVIEW

**Comment by 20001LastOrder** [comment]
@saminegash , @babasahebpinjar

**Review by saminegash — CHANGES_REQUESTED** [review]
there is also another conflict here btw

File: apps/slackbot/Dockerfile [hunk 1]
```diff
@@ -21,4 +21,4 @@ COPY . .
 EXPOSE 3000
 
 # Run the Flask app
-CMD ["python", "app.py"]
+CMD ["python", "bolt_app.py"]
```
[the remark this answers is not in the document]
Comment (20001LastOrder) [reply apps/slackbot/Dockerfile:24]: This is the new app that utilize pine cone. I didn’t change the other app.py file.

[the remark this answers is not in the document]
Comment (saminegash) [reply apps/slackbot/Dockerfile:24]: Can you resolve thw conflict then

[the remark this answers is not in the document]
Comment (20001LastOrder) [reply apps/slackbot/Dockerfile:24]: Done

**Review by saminegash — APPROVED** [review]

**Pull request merged**